### ISSUE TITLE
Vagrant init not working because of windows10 directory name...

### ISSUE THREAD

**Issue by diezcode** [mitchellh/vagrant#7516]
### Vagrant version

vagrant 1.8.4
### Host operating system

Windows 10
### Guest operating system

VirtualBox 5.0
### Vagrantfile

``` ruby
# Copy-paste your Vagrantfile here
```

no file ,error si coming when trying to create vagrant file : vagrant init
### Debug output

```
C:\dev\odoo_8_develop>vagrant init
C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:252:in `mkdir': Invalid argument @ dir_s_mkdir - C:/Users/St?phane Diez (Errno::EINVAL)
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:252:in `fu_mkdir'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:226:in `block (2 levels) in mkdir_p'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:224:in `reverse_each'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:224:in `block in mkdir_p'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:210:in `each'
        from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:210:in `mkdir_p'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:771:in `block in setup_home_path'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:766:in `each'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:766:in `setup_home_path'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:129:in `initialize'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/bin/vagrant:165:in `new'
        from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.8.4/bin/vagrant:165:in `<main>'

C:\dev\odoo_8_develop>
```
### Expected behavior

What should have happened?
vagrant init without errors.
### Actual behavior

What actually happened?
When writing "vagrant init" in the cmd it generates this error
### Steps to reproduce
1. Install vagrant on windows10 user with a è in the name
2. make a new folder and in the cmd write vagrant init
3. error shows up

**Comment by sethvargo** [comment]
Hi @diezcode 

Thank you for opening an issue, however, there is not enough information here to reproduce or provide a recommendation. When you first open the issue, a template pops up. That template provides instructions for getting your host and guest information, the version of Vagrant, and the debug output. Without that information, we will be unable to help.

Without that output, this error looks like you might have a special character in your path (such as a non-ascii character). Ruby, the underlying programming language Vagrant is written in, does not handle these situations ideally, especially on Windows. You can try setting `VAGRANT_HOME` to a different directory that does not include those special characters.

**Comment by gmkll** [comment]
I had the same error and the problem was setting VAGRANT_HOME (cft. setup_home_path in error stack). Setting it to a plain windows path fixed it for me (using cygwin).

**Comment by colinliactv** [comment]
Same error, on windows 10,vagrant 1.8.4
`Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'https://artifacts.company.com/artifactory/scm-images/centos/6.7/x86_64/20160329-1/centos-6.7-x86_64-20160329-1.virtualbox.box' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Box file was not detected as metadata. Adding it directly...
==> default: Adding box 'https://artifacts.company.com/artifactory/scm-images/centos/6.7/x86_64/20160329-1/centos-6.7-x86_64-20160329-1.virtualbox.box' (v0) for provider: virtualbox
    default: Downloading: https://artifacts.dev.activenetwork.com/artifactory/scm-images/centos/6.7/x86_64/20160329-1/centos-6.7-x86_64-20160329-1.virtualbox.box
    default: Progress: 100% (Rate: 249k/s, Estimated time remaining: --:--:--)
C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:252:in `mkdir': No such file or directory @ dir_s_mkdir - C:/Users/cli2/.vagrant.d/boxes/https-VAGRANTCOLON--VAGRANTSLASH--VAGRANTSLASH-artifacts.company.com-VAGRANTSLASH-artifactory-VAGRANTSLASH-scm-images-VAGRANTSLASH-centos-VAGRANTSLASH-6.7-VAGRANTSLASH-x86_64-VAGRANTSLASH-20160329-1-VAGRANTSLASH-centos-6.7-x86_64-20160329-1.virtualbox.box (Errno::ENOENT)
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:252:in `fu_mkdir'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:226:in `block (2 levels) in mkdir_p'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:224:in `reverse_each'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:224:in `block in mkdir_p'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:210:in `each'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/fileutils.rb:210:in `mkdir_p'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/pathname.rb:564:in `mkpath'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:160:in `block (3 levels) in add'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:448:in `with_temp_dir'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:133:in `block (2 levels) in add'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:448:in `with_temp_dir'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:114:in `block in add'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:435:in `block in with_collection_lock'
        from C:/Program Files (x86)/vagrant/embedded/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:434:in `with_collection_lock'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/box_collection.rb:103:in `add'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/box_add.rb:344:in `box_add'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/box_add.rb:136:in `add_direct'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/box_add.rb:110:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builder.rb:116:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/handle_box.rb:82:in `handle_box'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/handle_box.rb:42:in `block in call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/handle_box.rb:36:in `synchronize'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/handle_box.rb:36:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builder.rb:116:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builtin/call.rb:53:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/builder.rb:116:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/util/busy.rb:19:in `busy'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/action/runner.rb:66:in `run'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:225:in `action_raw'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:200:in `block in action'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/environment.rb:561:in `lock'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:186:in `call'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/machine.rb:186:in `action'
        from C:/Program Files (x86)/vagrant/embedded/gems/gems/vagrant-1.8.4/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'`

**Comment by dragon788** [comment]
@colinliactv  Could you please post an example of the Vagrantfile you are using? I have an idea that you are using ```vagrant init https://artifacts.company.com/artifactory/scm-images/centos/6.7/x86_64/20160329-1/centos-6.7-x86_64-20160329-1.virtualbox.box``` and this won't end well on Windows. You want to run ```vagrant init myboxname``` then edit the Vagrantfile and add a `config.vm.box_url = 'https://artifacts.company.com/artifactory/scm-images/centos/6.7/x86_64/20160329-1/centos-6.7-x86_64-20160329-1.virtualbox.box'` and see if that works for you. You are hitting a different error than the invalid character, yours is more similar to #7682 than this issue.

**Comment by diezcode** [comment]
The error is coming because of the é in my name, if I make a new user on windows with an e without signs, it works fine. The problem is that you can't change the name of your windows user directory.

**Comment by vertisan** [comment]
Same problem - Vagrant 1.8.5, Windows 10.
Problem is letter 'ł' in my username in Windows. This will be fixed?

**Comment by raoden1** [comment]
I'm sure that isn't a problem with thiss letter in your name, because I had the same letter in my name but when I changed it to antoher without special character it happend again.

I read somewhere that is problem with encoding, someone has with ASCII but is required non-ASCII

**Comment by raoden1** [comment]
[![success.png](http://s12.postimg.org/4oorl5arx/success.png)](http://postimg.org/image/iid4a73d5/)

in cmd better in shell type: setx VAGRANT_HOME "X:/your/path" or do it manually then type in shell like above in attached image. it works on win 10

**Comment by chrisroberts** [comment]
Fixed via #8385
